kernel/proc.c

```diff
@@ -486,7 +539,7 @@ sched(void)
 
   if(!holding(&p->lock))
     panic("sched p->lock");
-  if(mycpu()->noff != 1)
+  if(mycpu()->noff != 2)
```

Comment:
Why?

Reply:
Since I always hold proc_list_lock here and by default I hold proc->lock the number if locks is changed from 1 to 2